shared/nm-utils/nm-shared-utils.h

```diff
@@ -102,6 +102,7 @@ typedef struct {
 	union {
 		guint8 addr_ptr[1];
 		in_addr_t addr4;
+		struct in_addr addr4_struct;
```

Comment:
Why?

Also, if you actually need "struct in_addr", then you perhaps could drop the "in_addr_t" field, because it could be accessed via the "struct in_addr".

Reply:
Because I'll need to assign from `NMIPAddr` to a field of type `in_addr`:

```
+         *((struct sockaddr_in *) sockaddr) = (struct sockaddr_in) {
+              .sin_family = AF_INET,
+              .sin_addr   = addrbin.addr4_struct,
+              .sin_port   = htons (self->port),
+         };
```
Yes, I know it could be
```
+              .sin_addr   = { .s_addr = addrbin.addr4, },       
```

But would that be better? I don't understand which problem you see with the union. `NMIPAddr` main purpose is to be a union type, having another alias doesn't seem ugly to me.

Changing `addr4` from `in_addr_t` to `struct in_addr` will require a lot of changes throughout . In particular, because most of our internal API for IPv4 addresses is based on `in_addr_t` / `guint32` / `be32_t`, instead of `struct in_addr`. For example, `nm_utils_ip4_address_*()`. I would not object to refactor the code to prefer `struct in_addr` everywhere, but I don't mind either way. Anyway, not as part of this pull-request.